crates/fluvio-sc/src/services/auth/basic.rs

```diff
@@ -91,6 +91,58 @@ mod policy {
 
     type Role = String;
 
+    #[derive(Debug, Clone, Eq, PartialEq)]
+    pub struct ActionUrn {
+        pub action: Action,
+        pub instances: Option<Vec<String>>,
+    }
+
+    impl ActionUrn {
+        pub fn new(action: Action, instances: Option<Vec<String>>) -> Self {
+            Self { action, instances }
+        }
+    }
+
+    impl Serialize for ActionUrn {
+        fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: serde::Serializer,
+        {
+            let action_str =
+                serde_json::to_string(&self.action).map_err(serde::ser::Error::custom)?;
+            let urn = match &self.instances {
+                Some(instances) => {
+                    format!("{}:{}", action_str.trim_matches('"'), instances.join(":"))
+                }
+                None => action_str.trim_matches('"').to_string(),
+            };
+            serializer.serialize_str(&urn)
+        }
+    }
+
+    impl<'de> serde::Deserialize<'de> for ActionUrn {
+        fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where
+            D: serde::Deserializer<'de>,
+        {
+            use serde::de::Error;
+            let urn = String::deserialize(deserializer)?;
+            let parts: Vec<&str> = urn.split(':').collect();
+
+            let action_str = parts.first().ok_or(Error::custom("missing action"))?;
+            let action = serde_json::from_str(format!("\"{}\"", action_str).as_str())
+                .map_err(Error::custom)?;
+
+            let instances = if parts.len() > 1 {
+                Some(parts[1..].iter().map(|s| s.to_string()).collect())
+            } else {
+                None
+            };
+
+            Ok(Self { action, instances })
```

Comment:
please add test for serialize and deserialize

Reply:
The `test_policy_serialization` are serializing and deserializing it, does not cover it? 

Actually, this change was made first on that PR: https://github.com/infinyon/fluvio/pull/4021/files

Reply:
Hmm.  I didn't see updated file or new file with changes

Reply:
https://github.com/infinyon/fluvio/pull/4021/files#diff-3542a6ea8dd07337a7a2bce491d21abb9c32597701d4ff6f3936e73293ead019R338

I added `test_action_urn_serialization` and `test_action_urn_deserialization` to test directly now anyway.